src/server/service/batch.rs

```diff
@@ -83,60 +93,124 @@ impl ReqBatcher {
     }
 }
 
+pub struct BatcherBuilder {
+    pool_size: usize,
+    enable_batch: bool,
+}
+
+impl BatcherBuilder {
+    pub fn new(enable_batch: bool, pool_size: usize) -> Self {
+        BatcherBuilder {
+            enable_batch,
+            pool_size,
+        }
+    }
+    pub fn build(&self, queue_per_worker: usize, req_batch_size: usize) -> Option<ReqBatcher> {
+        if !self.enable_batch {
+            return None;
+        }
+        if req_batch_size > self.pool_size * MIN_BATCH_GET_REQUEST_COUNT
+            && queue_per_worker >= MIN_BATCH_GET_REQUEST_COUNT
+        {
+            return Some(ReqBatcher::new(req_batch_size / self.pool_size));
+        }
+        if req_batch_size >= MIN_BATCH_GET_REQUEST_COUNT
+            && queue_per_worker >= MAX_QUEUE_SIZE_PER_WORKER
+        {
+            return Some(ReqBatcher::new(req_batch_size));
```

Comment:
`req_batch_size != get_req_batch_size`, I'm not sure how much the overhead though for delaying a few gets among lots of writes. If it's worth considering, maybe we could fix it by using `if filtered_request_count >= self.target_batch_size` as commit condition, where filtered requests aren't type sensitive.

Reply:
Yes.... The batch may be slow than the best size.